recipes/full_finetune_distributed.py

```diff
@@ -618,6 +707,56 @@ def _setup_data(
 
         return sampler, dataloader
 
+    def save_checkpoint_async(
```

Comment:
can we make it an utility?

```
save_checkpoint_async(
	epoch = epoch,
	model_state_dict = self._model.state_dict(),
	optim_state_dict = self._optim_ckpt_wrapper.state_dict() if self._optimizer_in_bwd else self._optimizer.state_dict(),
	checkpointer = self.dcp_checkpointer
	is_rank_zero = self._is_rank_zero
)
```

Reply:
This may not be relevant with the latest refactor. CheckpointClient will be the entry-point for the checkpointing APIs across recipes. This refactor is in line with how we do it with other trainers.